hw2/script.py

```diff
@@ -0,0 +1,51 @@
+# Разработать систему регистрации пользователя, используя Pydantic для валидации входных данных,
+# обработки вложенных структур и сериализации. Система должна обрабатывать данные в формате JSON.
+from hw2.models.user import User
+
+
+# 1. Создать классы моделей данных с помощью Pydantic для пользователя и его адреса.
+
+# 2. Реализовать функцию, которая принимает JSON строку, десериализует её в объекты Pydantic,
+# валидирует данные, и в случае успеха сериализует объект обратно в JSON и возвращает его.
+def persist_data(data: str) -> str|None:
+    try:
+        user = User.model_validate_json(data)
+        return user.model_dump_json(indent=4)
+    except ValueError as e:
+        print(e)
+        return None
```

Comment:
Обычно стараются обрабатывать только ту строку, где ожидают появление ошибки. Строка с return у вас может быть размещена в блоке `else`
```suggestion
    try:
        user = User.model_validate_json(data)
    except ValueError as e:
        print(e)
    else:
        return user.model_dump_json(indent=4)    
```

Reply:
Поправил